### PR TITLE
[CI][Intel] enable MetisApplication

### PR DESCRIPTION
**Pull request by philbucher**
**Description**
This enables the MetisApp in the CI that uses the IntelCompiler

I am adding it first as it is needed for many MPI tests in the Core (for the partitioning)

### CODE REVIEW

**Comment by philbucher** [comment]
this is the error:
~~~
double free or corruption (out)
double free or corruption (out)
[5f48cec15944:01907] *** Process received signal ***
[5f48cec15944:01907] Signal: Aborted (6)
[5f48cec15944:01907] Signal code:  (-6)
[5f48cec15944:01906] *** Process received signal ***
[5f48cec15944:01906] Signal: Aborted (6)
[5f48cec15944:01906] Associated errno: Unknown error 32766 (32766)
[5f48cec15944:01906] Signal code:  (-1350335440)
[5f48cec15944:01907] [ 0] [5f48cec15944:01906] /lib/x86_64-linux-gnu/libc.so.6(+0x46210)[0x7f07bbb52210]
[5f48cec15944:01907] [ 1] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb)[0x7f07bbb5218b]
[5f48cec15944:01907] [ 2] /lib/x86_64-linux-gnu/libc.so.6(abort+0x12b)[0x7f07bbb31859]
[5f48cec15944:01907] [ 3] /lib/x86_64-linux-gnu/libc.so.6(+0x903ee)[0x7f07bbb9c3ee]
[5f48cec15944:01907] [ 4] /lib/x86_64-linux-gnu/libc.so.6(+0x9847c)[0x7f07bbba447c]
[5f48cec15944:01907] [ 5] /lib/x86_64-linux-gnu/libc.so.6(+0x9a120)[0x7f07bbba6120]
[5f48cec15944:01907] [ 6] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(_ZN9__gnu_cxx13new_allocatorIdE10deallocateEPdm+0x20)[0x7f07b8cfdd80]
[5f48cec15944:01907] [ 7] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(+0xd4ec4b)[0x7f07b8db9c4b]
[5f48cec15944:01907] [ 8] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(+0x11d7de0)[0x7f07b9242de0]
[5f48cec15944:01907] [ 9] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(+0x11d99f7)[0x7f07b92449f7]
[5f48cec15944:01907] [10] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(+0x11d9138)[0x7f07b9244138]
[5f48cec15944:01907] [11] [ 0] /lib/x86_64-linux-gnu/libc.so.6(+0x46210)[0x7f6428fcd210]
[5f48cec15944:01906] [ 1] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb)[0x7f6428fcd18b]
[5f48cec15944:01906] [ 2] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(+0x11d9e24)[0x7f07b9244e24]
[5f48cec15944:01907] [12] /lib/x86_64-linux-gnu/libc.so.6(abort+0x12b)[0x7f6428fac859]
[5f48cec15944:01906] [ 3] /lib/x86_64-linux-gnu/libc.so.6(+0x903ee)[0x7f64290173ee]
[5f48cec15944:01906] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(_ZNSt5arrayIN5boost7numeric5ublas6vectorINS2_6matrixIdNS2_15basic_row_majorImlEENS2_15unbounded_arrayIdSaIdEEEEENS7_ISA_SaISA_EEEEELm10EED1Ev+0x55)[0x7f07b8e6df45]
[5f48cec15944:01907] [13] [ 4] /lib/x86_64-linux-gnu/libc.so.6(+0x9847c)[0x7f642901f47c]
[5f48cec15944:01906] [ 5] /lib/x86_64-linux-gnu/libc.so.6(+0x9a120)[0x7f6429021120]
[5f48cec15944:01906] [ 6] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(_ZN9__gnu_cxx13new_allocatorIdE10deallocateEPdm+0x20)[0x7f6426178d80]
/__w/Kratos/Kratos/bin/Custom/libs/libKratosCore.so(_ZN6Kratos30GeometryShapeFunctionContainerINS_12GeometryData17IntegrationMethodEED1Ev+0x48)[0x7f07b4cfc7d8]
[5f48cec15944:01907] [14] /__w/Kratos/Kratos/bin/Custom/libs/libKratosCore.so(_ZN6Kratos12GeometryDataD1Ev+0x32)[0x7f07b4cfd4ae]
[5f48cec15944:01907] [15] /lib/x86_64-linux-gnu/libc.so.6(+0x49a27)[0x7f07bbb55a27]
[5f48cec15944:01906] [5f48cec15944:01907] [ 7] [16] /lib/x86_64-linux-gnu/libc.so.6(on_exit+0x0)[0x7f07bbb55be0]
[5f48cec15944:01907] [17] /usr/bin/python3[0x67cbdb]
[5f48cec15944:01907] [18] /usr/bin/python3[0x67cbfb]
[5f48cec15944:01907] [19] /usr/bin/python3(PyErr_PrintEx+0x16)[0x67ce26]
[5f48cec15944:01907] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(+0xd4ec4b)[0x7f6426234c4b]
[5f48cec15944:01906] [ 8] [20] /usr/bin/python3(PyRun_SimpleFileExFlags+0x1ac)[0x67da9c]
[5f48cec15944:01907] [21] /usr/bin/python3(Py_RunMain+0x212)[0x6b6132]
[5f48cec15944:01907] [22] /usr/bin/python3(Py_BytesMain+0x2d)[0x6b64bd]
[5f48cec15944:01907] [23] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3)[0x7f07bbb330b3]
/__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(+0x11d7de0)[0x7f64266bdde0]
[5f48cec15944:01906] [ 9] [5f48cec15944:01907] [24] /usr/bin/python3(_start+0x2e)[0x5f927e]
[5f48cec15944:01907] *** End of error message ***
/__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(+0x11d99f7)[0x7f64266bf9f7]
[5f48cec15944:01906] [10] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(+0x11d9138)[0x7f64266bf138]
[5f48cec15944:01906] [11] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(+0x11d9e24)[0x7f64266bfe24]
[5f48cec15944:01906] [12] /__w/Kratos/Kratos/bin/Custom/libs/Kratos.cpython-38-x86_64-linux-gnu.so(_ZNSt5arrayIN5boost7numeric5ublas6vectorINS2_6matrixIdNS2_15basic_row_majorImlEENS2_15unbounded_arrayIdSaIdEEEEENS7_ISA_SaISA_EEEEELm10EED1Ev+0x55)[0x7f64262e8f45]
[5f48cec15944:01906] [13] /__w/Kratos/Kratos/bin/Custom/libs/libKratosCore.so(_ZN6Kratos30GeometryShapeFunctionContainerINS_12GeometryData17IntegrationMethodEED1Ev+0x48)[0x7f64221777d8]
[5f48cec15944:01906] [14] /__w/Kratos/Kratos/bin/Custom/libs/libKratosCore.so(_ZN6Kratos12GeometryDataD1Ev+0x32)[0x7f64221784ae]
[5f48cec15944:01906] [15] /lib/x86_64-linux-gnu/libc.so.6(+0x49a27)[0x7f6428fd0a27]
[5f48cec15944:01906] [16] /lib/x86_64-linux-gnu/libc.so.6(on_exit+0x0)[0x7f6428fd0be0]
[5f48cec15944:01906] [17] /usr/bin/python3[0x67cbdb]
[5f48cec15944:01906] [18] /usr/bin/python3[0x67cbfb]
[5f48cec15944:01906] [19] /usr/bin/python3(PyErr_PrintEx+0x16)[0x67ce26]
[5f48cec15944:01906] [20] /usr/bin/python3(PyRun_SimpleFileExFlags+0x1ac)[0x67da9c]
[5f48cec15944:01906] [21] /usr/bin/python3(Py_RunMain+0x212)[0x6b6132]
[5f48cec15944:01906] [22] /usr/bin/python3(Py_BytesMain+0x2d)[0x6b64bd]
[5f48cec15944:01906] [23] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3)[0x7f6428fae0b3]
[5f48cec15944:01906] [24] /usr/bin/python3(_start+0x2e)[0x5f927e]
[5f48cec15944:01906] *** End of error message ***
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpiexec noticed that process rank 0 with PID 0 on node 5f48cec15944 exited on signal 6 (Aborted).
--------------------------------------------------------------------------
Completed KratosMPICore tests with exit code 1!
~~~

I don't know yet where exactly it happens (I think in `test_mpi_communicator.py`)

Also I cannot reproduce it locally with pretty much the same setup :/

**Comment by philbucher** [comment]
so it seems that as soon as I use a ModelPart that is filled, then the test start to fail :/

Any idea how to proceed @KratosMultiphysics/technical-committee?

**Comment by philbucher** [comment]
after some more testing I see that simply importing the MetisApp causes the double-free error

any idea how to proceed?
@roigcarlo ?

**Comment by roigcarlo** [comment]
Mm not really, let me take a look...

File: applications/MetisApplication/tests/test_MetisApplication.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def AssembleTestSuites():
     # - testSmallExample
     smallSuite = suites['small']
     smallSuite.addTest(KratosMetisGeneralTests('testSmallExample'))
-    smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([TestMetisSubModelPartList]))
+    # smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([TestMetisSubModelPartList]))
```
Comment (RiccardoRossi) [inline applications/MetisApplication/tests/test_MetisApplication.py:32]: why deactivating this?

**Review by RiccardoRossi — COMMENTED** [review]
@philbucher  should this be kept on hold until te tests are on?

File: applications/MetisApplication/tests/test_MetisApplication.py (hunk 1 above)
Comment (philbucher) [reply applications/MetisApplication/tests/test_MetisApplication.py:32]: I am trying to see where the problem was
Turns out, as posted above, that simply importing the MetisApp causes a segfault at the end of the execution

**Comment by philbucher** [comment]
> @sunethwarna will pass it in FullDebug into valgrind to see if we get some hint about the error.

It already runs in `FullDebug` in the CI. Thats the first thing I checked

**Comment by RiccardoRossi** [comment]
one possible point is that pybind11 always had problems with intel. I wonder if it is related to this

**Pull request closed without merging**